community/kernel/src/main/java/org/neo4j/kernel/impl/api/LockingStatementOperations.java

```diff
@@ -229,17 +229,9 @@ public long nodeCreate( KernelStatement statement )
     public long relationshipCreate( KernelStatement state, int relationshipTypeId, long startNodeId, long endNodeId )
             throws EntityNotFoundException
     {
-        // Order the locks to lower the risk of deadlocks with other threads adding rels concurrently
-        if(startNodeId < endNodeId)
-        {
-            state.locks().acquireExclusive( ResourceTypes.NODE, startNodeId );
-            state.locks().acquireExclusive( ResourceTypes.NODE, endNodeId );
-        }
-        else
-        {
-            state.locks().acquireExclusive( ResourceTypes.NODE, endNodeId );
-            state.locks().acquireExclusive( ResourceTypes.NODE, startNodeId );
-        }
+        state.locks().acquireShared( ResourceTypes.SCHEMA, schemaResource() );
```

Comment:
Oh, I don't know how that snuck in there. That would change behaviour, something we don't want to do for an older version.